code/modules/mining/boulder_processing/_boulder_processing.dm

```diff
@@ -0,0 +1,361 @@
+/obj/machinery/bouldertech
+	name = "bouldertech brand refining machine"
+	desc = "You shouldn't be seeing this! And bouldertech isn't even a real company!"
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "ore_redemption"
+	anchored = TRUE
+	density = TRUE
+
+	/// What is the efficiency of minerals produced by the machine?
+	var/refining_efficiency = 1
+	/// How many boulders can we process maximum per loop?
+	var/boulders_processing_max = 1
+	/// What boulder(s) are we holding?
+	var/list/boulders_contained = list()
+	/// How many boulders can we hold maximum?
+	var/boulders_held_max = 1
+	/// Does this machine have a mineral storage link to the silo?
+	var/holds_minerals = FALSE
+	/// What materials do we accept and process out of boulders? Removing iron from an iron/glass boulder would leave a boulder with glass.
+	var/list/processable_materials = list()
+	/// If we get a boulder with one of these inside, we'll drop it back out.
+	var/static/list/drop_if_contained = list(
+		/obj/item/relic,
+	)
+
+	/// What sound plays when a thing operates?
+	var/usage_sound = 'sound/machines/mining/wooping_teleport.ogg'
+	/// Cooldown associated with the usage_sound played.
+	COOLDOWN_DECLARE(sound_cooldown)
+
+	/// Silo link to it's materials list.
+	var/datum/component/remote_materials/silo_materials
+	///Does this machine hold mining points?
+	var/holds_mining_points = FALSE
+	/// Mining points held by the machine for miners.
+	var/points_held = 0
+
+/obj/machinery/bouldertech/Initialize(mapload)
+	. = ..()
+	register_context()
+	if(holds_minerals)
+		silo_materials = AddComponent(
+			/datum/component/remote_materials, \
+			mapload, \
+			mat_container_flags = BREAKDOWN_FLAGS_ORM|MATCONTAINER_NO_INSERT|MATCONTAINER_EXAMINE \
+		)
+
+/obj/machinery/bouldertech/LateInitialize()
+	. = ..()
+	if(!holds_minerals)
+		return
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
```

Comment:
Why is this in `LateInitialize()` ? this means boulder tech machinery constructed by players in game won't have this component since `LateInitialize()` never gets called after mapload. Is this intended?

Reply:
Late Init is called on atom init post mapload it would seem, in testing this does appear to work properly

Reply:
Well what I meant to say was after the map is fully loaded then Late Init get's called, Once for all atoms and that's it never again.

Have you tried spawning this machine in game as an admin and check if this component exists in it's VV menu?

I don't think it will be there though.